src/transformStateWithClones.js

```diff
@@ -7,7 +7,42 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const stateClones = [
+    {
+      ...state,
+    },
+  ];
+
+  for (let i = 0; i < actions.length; i++) {
+    if (i !== 0) {
+      stateClones[i] = Object.assign({}, stateClones[i - 1]);
+    }
+
```

Comment:
you can make a destructuring from `stateClones[i]`
Also, you don't need index here, so, do it with `for of` loop